### PR TITLE
refactor: split nodes & runtimes

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

Moves `rpc` crate to the root of the directory and adds a separate `standalone` node / runtime.

### CODE REVIEW

File: parachain/Cargo.toml [hunk 1]
```diff
@@ -22,54 +22,74 @@ serde = { version = "1.0.119", features = ["derive"] }
...
+sp-consensus = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
 sp-consensus-aura = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
+sp-session = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
+sp-offchain = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
+sp-blockchain = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
+sp-block-builder = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
+sp-keystore = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
+sp-timestamp = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
+sp-trie = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
+sp-transaction-pool = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
```
Comment (sander2) [inline parachain/Cargo.toml:62]: Why are so many of these "sp-*" dependencies added?

File: parachain/Cargo.toml [hunk 2]
```diff
@@ -22,54 +22,74 @@ serde = { version = "1.0.119", features = ["derive"] }
...
 frame-benchmarking = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
 frame-benchmarking-cli = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.7" }
 
+# RPC related dependencies
+jsonrpc-core = "15.1.0"
+
 # Cumulus dependencies
-cumulus-primitives = { package = "cumulus-primitives-core", git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.7", optional = true }
-cumulus-service = { package = "cumulus-client-service", git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.7", optional = true }
+cumulus-client-cli = { git = "https://github.com/paritytech/cumulus", branch = "polkadot-v0.9.7" }
```
Comment (sander2) [inline parachain/Cargo.toml:74]: Is there any way to extract the cumulus stuff into a separate package, that won't be compiled by the standalone version?

**Review by sander2 — COMMENTED** [review]
Is there any way to prevent the new code duplication? E.g. cli.rs, command.rs

File: parachain/Cargo.toml (hunk 1 above)
Comment (gregdhill) [reply parachain/Cargo.toml]: Mostly because I removed the `service` crate, I've just gone through and removed all unused dependencies here.

**Comment by gregdhill** [comment]
Wrt: duplication, we can probably extract some common components in the future but it's mostly boilerplate and there are some custom commands in each.

**Review by sander2 — APPROVED** [review]

**Pull request merged**